lib/app.js

```diff
@@ -9,12 +9,13 @@ const ErrorDebug = process.env.NODE_ENV === 'production'
 
 export default class App extends Component {
   static childContextTypes = {
-    headManager: PropTypes.object
+    headManager: PropTypes.object,
+    router: PropTypes.object
   }
 
   getChildContext () {
-    const { headManager } = this.props
-    return { headManager }
+    const { headManager, router } = this.props
```

Comment:
I think we should not expose the whole router API. But only the fields we need.

We can use the same logic used inside `lib/router/index.js`
May be I can work on this.